### ISSUE TITLE
message: "openrct.exe - Entry Point Not Found" (OPENRCT2 NOT OPENING) 

### ISSUE THREAD

**Issue by jamtraks** [OpenRCT2/OpenRCT2#6741]
OS: Vista 32-bit
Version: Launcher Version: 0.0.6 x86 or any recent openrct2 version
Commit/Build: OpenRCT2 Build: 3112, OpenRCT2 Git Hash: a3be15c

I had been without internet access for weeks before today and the openrct2 version i was using during that time worked well enough, and i could play, (though it did freeze once in a while before loading). Now today after my launcher has installed the latest "develop" version... i cannot open openrct2 at all, i get this message: openrct2.exe - Entry Point Not Found (x) The procedure entry point K32GetModuleFileNameExW could not be located in the dynamic link library KERNEL32.dll

    [did not test ] Reproducible in RCT2 (vanilla)?
    [n/a ] Multiplayer?

Steps to reproduce:

    press "play" button after opening launcher

Dump file
<none created

Screenshots / Video:
![openrct2 kernel error](https://user-images.githubusercontent.com/17103908/33286090-736566fe-d382-11e7-87b5-d563406a971a.jpg)

Save game:
n/a

**Comment by Gymnasiast** [comment]
This entry point should be present in Vista with SP1. Make sure you have all Service Packs and updates installed. Or even better, upgrade to a supported operating system.

**Comment by rwjuk** [comment]
@jamtraks Do you have Service Pack 1 installed for Windows Vista? If not, download and install that from here: https://www.microsoft.com/en-gb/download/details.aspx?id=30 and then retry.

OpenRCT2 requires SP1 to work.

**Comment by jamtraks** [comment]
Okay... well, i attempted to install (service pack one) from this link and i got the message "service pack one already installed". My computer is telling me that i have service pack two installed. Like i said, there was no problem running the openrct2 version from several weeks ago, this is a new issue that i did not have before. I am guessing the easiest thing i could do on my end is buy a windows 7, 8, or 10 computer, but i was not planning to any time soon.

**Comment by Patrik356b** [comment]
Perhaps you could take a look at Linux?
Some of the different flavors have the option to run without being installed, so you could try running OpenRCT2 on Linux and see if it works better for you.

**Comment by Gymnasiast** [comment]
Linux is a very interesting route. But if it's not an option, I would suggest you downgrade to release 0.1.1. In that case, you can also get rid of the launcher and just install the appropriate version from OpenRCT2.org.

**Comment by jamtraks** [comment]
I don't see where to download openrct2 versions older than"5 days". Could someone explain to me how i might acquire a version older than 5 days?

-and, no i don't think i could run linux on this computer...
-Why was openrct2 working before my newest update? Am i the only person experiencing this problem? Can't you all just fix what's causing this? It wasn't a problem in versions older than about 8-16 weeks ago...

**Comment by Gymnasiast** [comment]
It's probably due to using a newer version of MS Visual Studio, as we don't call that entry point ourselves.

I reported the 404 you got to the maintainer of OpenRCT2.org.

**Comment by janisozaur** [comment]
@IntelOrca  https://github.com/discordapp/discord-rpc/pull/68

**Comment by jamtraks** [comment]
at openrct2.org i was told:
"As we already said on Github, it was caused by Discord integration. Janisozaur actually sent in a patch to allow for Vista support, which has been merged, but we need to update our dependencies. After that, the game should work on Vista again."

**Does someone have an idea of how long this will take?**

**Comment by Patrik356b** [comment]
A day, a week, something?
We don't know as this project is developed during free time.

**Comment by janisozaur** [comment]
@IntelOrca https://github.com/discordapp/discord-rpc/releases/tag/v2.1.0

**Comment by jamtraks** [comment]
Is there a way i can use this zip file to play? I don't know exactly what to do with this.

**Comment by IntelOrca** [comment]
@jamtraks there isn't anything you can do until we update the game with the new version of this library.

In the meantime, you can play an older build prior to the integration:
https://openrct2.org/downloads/develop/891de0a

**Comment by AaronVanGeffen** [comment]
> @IntelOrca https://github.com/discordapp/discord-rpc/releases/tag/v2.1.0

Will the problem be resolved when updating to this v2.1.0? It looks like there hasn't been any release in the mean time.

**Comment by janisozaur** [comment]
> Will the problem be resolved when updating to this v2.1.0?

yes

**Comment by Gymnasiast** [comment]
We just made fixes. Develop builds should now work on Vista again. Please update and verify that it works.

**Comment by jamtraks** [comment]
The game _is_ opening. However, if i try opening the game with the normal openrct2 icon and not the launcher, it is not responding and sometimes opens after about 4 minutes and sometimes doesn't. Also when i first tried opening the launcher after it updated, it took about four minutes to actually start the title screen. This was an issue for me before i created this here "topic", and still seems to be an issue. I just now tried closing and reopening openrct2 through the launcher, and it is stalling again. This is still a problem, i am sometimes having to wait four minutes for the title screen to appear. When this used to happen months ago, my settings such as cheats and options would sometimes "reset" to default when this happens, and it seemed to happen randomly.
I also get an elevated permissions message, before the game begins to launch.

![openrct2 not responding](https://user-images.githubusercontent.com/17103908/37235324-b7f240fa-23cb-11e8-8250-96848c0f2a67.jpg)

![openrct2 elevatedpermissions](https://user-images.githubusercontent.com/17103908/37235330-c1d44348-23cb-11e8-8235-580d1c78d99c.jpg)

**Comment by AaronVanGeffen** [comment]
If you're running the game from a hard drive, and have a lot of custom content installed, it might take a long time to index everything… But 4 minutes is really stretching it — I'd say that's not normal.

For the record, are you running the game as an administrator? That is, is the permission warning you're getting appropriate?

**Comment by jamtraks** [comment]
Yes i am the administrator, on a laptop with permissions and user options, so i just press "ok" and the game starts, but like i said it often takes literally 4 minutes to actually start, yet sometimes it just starts in about 3 seconds, it is quite random, the way this occurs.

**Comment by jamtraks** [comment]
Yes it's still stalling when i try to open.

**Comment by IntelOrca** [comment]
Like @AaronVanGeffen said, when you launch the game, OpenRCT2 will create an index of all the objects, scenarios and tracks you have installed. This is done only when we change something related to those indexes or when you add new content either downloaded from the internet or by opening a park that has packed objects. It will take a long time if you have lots of custom content and I expect the length of time is due to the age of your computer, hardware and operating system.

Seeing as you can now open the game this particular issue can be closed.

**Comment by jamtraks** [comment]
alright... but like i said it is still stalling

**Comment by janisozaur** [comment]
It's easy to check if it seems stalling while processing the custom data: when processing, we output the progress on stdout ( https://github.com/OpenRCT2/OpenRCT2/pull/6731 ), so just check if there is anything being reported there.

**Comment by Gymnasiast** [comment]
In layman's terms: if you start OpenRCT2 from the command prompt, it will output the progress of object indexing.

You probably have a slow system and a lot of objects, hence why it takes 4 minutes.
